packages/edit-site/src/components/page-templates/dataviews-templates.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * External dependencies
+ */
+import removeAccents from 'remove-accents';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalHeading as Heading,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useState, useMemo, useCallback } from '@wordpress/element';
+import { useEntityRecords } from '@wordpress/core-data';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * Internal dependencies
+ */
+import Page from '../page';
+import Link from '../routes/link';
+import AddedBy from '../list/added-by';
+import { TEMPLATE_POST_TYPE } from '../../utils/constants';
+import { DataViews } from '../dataviews';
+import {
+	useResetTemplateAction,
+	useTrashTemplateAction,
+} from './template-actions';
+
+const EMPTY_ARRAY = [];
+const defaultConfigPerViewType = {
+	list: {},
+	grid: {},
+};
+
+const DEFAULT_VIEW = {
+	type: 'list',
+	search: '',
+	page: 1,
+	perPage: 5, //20,
+	// All fields are visible by default, so it's
+	// better to keep track of the hidden ones.
+	hiddenFields: [],
+	layout: {},
+};
+
+function normalizeSearchInput( input = '' ) {
+	return removeAccents( input.trim().toLowerCase() );
+}
+
+export default function DataviewsTemplates() {
+	const [ view, setView ] = useState( DEFAULT_VIEW );
+
+	const { records: allTemplates, isResolving: isLoadingData } =
+		useEntityRecords( 'postType', TEMPLATE_POST_TYPE, {
+			per_page: -1,
+		} );
+	const { shownTemplates, paginationInfo } = useMemo( () => {
+		if ( ! allTemplates )
```

Comment:
I'd add the parenthesis here, so it's obvious what code belongs in this condition and there's no subtle bugs.